src/Gateways/AlipayHK/Support.php

```diff
@@ -13,14 +13,40 @@
 use Yansongda\Pay\Exceptions\InvalidSignException;
 use Yansongda\Pay\Log;
 use Yansongda\Supports\Collection;
+use Yansongda\Supports\Str;
 
 class Support extends \Yansongda\Pay\Gateways\Alipay\Support
 {
-    public static function generateSign($parmas, $privateKey = null): string
+    /**
+     * @param $parmas
+     * @param null $privateKey
+     * @param string $mode
+     * @param string $separator // While generating RSA Sign for App Payment, $separator should be "
+     *
+     * For Example _input_charset=UTF-8 [X] if pass this to Android SDK, It will return System Busy error.
+     *             _input_charset="UTF-8" [V] every value should wrap with "" and then sign with RSA, Sdk will become normal
+     *
+     * However, doing query with RSA Sign, the parameter value should not be wrapped with "
+     *
+     * For Example _input_charset="UTF-8" [X] it will return invalud sign error from alipay query api
+     *             _input_charset=UTF-8 [V] it is good
+     * @return string
+     */
+    public static function generateSign($parmas, $privateKey = null, $mode = "MD5", $separator = ""): string
     {
         ksort($parmas);
 
-        return md5(urldecode(http_build_query($parmas)) . $privateKey);
+        $sign = "";
+
+        if ($mode == "MD5") {
+            $sign = md5(urldecode(http_build_query($parmas)) . $privateKey);
+        }
+
+        if ($mode == "RSA") {
+            $sign = self::generateRSASign($parmas, $privateKey, $separator);
+        }
+
+        return $sign;
```

Comment:
since the method was used to generate the signature by md5 encryption, it never ideal to exists here with another static method. just extract it, it would be more clean and readable.